### PR TITLE
v3.34.100 — STRK-123: trade linking

### PR DESCRIPTION
**Pull request by lbruton**
## Summary
- Adds optional trade linking for traded dispositions, including bidirectional `tradedForUuids` / `tradedFromUuid` relationships and editable link management from disposed or received item views.
- Adds trade value metadata, spot-derived defaults, trade gain/loss labeling, received-item provenance, and graceful fallback handling for unlinked, missing, re-disposed, and restored links.
- Extends persistence surfaces for trade links across ZIP JSON backup, standalone CSV import/export, change log, diff/hash surfaces, and core Playwright coverage.

## Links
- Plane: STRK-123
- Sketch: `DocVault/Projects/StakTrakr/sketches/STRK-123-trade-linking/`
- Version: `3.34.100`

## Test Plan
- [x] `npm run lint` — passes with 2 existing unused-disable warnings in `js/diff-engine.js` and `js/diff-modal.js`
- [x] `npm test` — 146 passed
- [x] `npm run test:unit` — 120 passed
- [x] `npx playwright test tests/playwright/core/smoke.spec.js` — 12 passed after version bump
- [x] Focused trade regression: `npx playwright test tests/playwright/core/disposition.spec.js tests/playwright/core/import-export.spec.js tests/playwright/core/attachments-cloud.spec.js --grep "trade"` — 6 passed
- [x] Codacy CLI changed-file scan: OpenGrep 0 findings; Trivy reported two pre-existing poller dependency CVEs outside this PR scope; Lizard/CLI ESLint produced existing repo noise, no new critical/high issue accepted into scope

## Test Inventory Delta
- `+6 -0` tests
- `+0 -0` test files
- Updated `tests/playwright/coverage-map.csv`

## Notes
- `/update-spot-bundle` ran with `SQLD_URL=http://192.168.1.81:8080`; data was already current through 2026-05-26 and the rebuilt bundle had no git delta.
- DocVault Data Model and STRK-123 sketch docs were updated and pushed separately to DocVault `main`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ---
 
+## [3.34.100] - 2026-05-26
+
+### Changed — STRK-123: Trade linking
+
+- **Trade provenance and value tracking**: Added bidirectional trade links between disposed traded items and received inventory, with editable trade relationships, spot-derived trade values, provenance display, undo support, CSV/ZIP backup coverage, and cloud-sync hash visibility. (STRK-123)
+
+---
```
[the remark this answers is not in the document]
Comment (lbruton) [reply CHANGELOG.md]: **False positive — already addressed.** `/update-spot-bundle` was run as documented in the PR description. The `data/spot-history-bundle.js` and `data/spot-history-2026.json` files are current through 2026-05-26. The rebuilt bundle had no git delta because the underlying sqld data was already up-to-date; this is expected behavior, not a missed step.

File: js/inventory.js [hunk 2]
```diff
@@ -849,6 +852,113 @@ const disposeItem = (idx) => {
...
+  }
+  const receivedItem = typeof findItemByUuid === "function" ? findItemByUuid(receivedUuid) : null;
+  if (receivedItem?.tradedFromUuid === disposedItem.uuid) delete receivedItem.tradedFromUuid;
+  if (log) {
+    pushTradeLinkChange(disposedItem, receivedItem, before, {
+      disposedUuid: disposedItem.uuid,
+      receivedUuid,
+      action: "unlink",
+    });
+  }
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/inventory.js:907]: **Fixed in `97b98c59`** — `removeTradeLinkReference` now captures `tradedFromUuid` in the `before` snapshot (pre-unlink state of the received item) and includes `tradedForUuids` plus `tradedFromUuid: null` in `newValue`. Undo correctly restores back-references and redo no longer wipes sibling trade links on the disposed item.

File: js/inventory.js [hunk 3]
```diff
@@ -784,6 +784,9 @@ const openRemoveItemModal = (idx, preDispose = false) => {
   if (notesInput) notesInput.value = "";
   const amountGroup = safeGetElement("dispositionAmountGroup");
   if (amountGroup) amountGroup.style.display = "";
+  window.resetPendingTradeLinks?.();
+  const tradeSection = safeGetElement("tradeLinkSection");
+  if (tradeSection) tradeSection.style.display = "none";
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/inventory.js:794]: **Stale — no longer applicable.** The 'Edit Trade' flow was reworked under STRK-128/STRK-132. It no longer opens the dispose modal to reconstruct trade links from scratch; instead, an inline editor in `viewModal.js` pre-populates existing links and lets the user add/remove chips without losing them.

File: js/inventory.js [hunk 4]
```diff
@@ -849,6 +852,113 @@ const disposeItem = (idx) => {
...
+      receivedUuid,
+      tradedForUuids: [...disposedItem.disposition.tradedForUuids],
+      tradedFromUuid: receivedItem.tradedFromUuid || null,
+    };
+    if (!disposedItem.disposition.tradedForUuids.includes(receivedUuid)) {
+      disposedItem.disposition.tradedForUuids.push(receivedUuid);
+    }
+    const tradeValue =
+      typeof computeTradeValue === "function" ? computeTradeValue(receivedItem, tradeDate) : null;
+    if (tradeValue) disposedItem.disposition.tradeValues[receivedUuid] = tradeValue;
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/inventory.js:944]: **Stale — no longer applicable.** The `.trade-value-input` UI was removed during the STRK-124 polish pass. Current `renderPendingTradeLinks` uses chip pills without per-item value inputs. The disposition Amount field is the single editable trade value, and STRK-132 cost basis carryover uses the given-up item's melt value at trade date.

File: js/viewModal.js [hunk 5]
```diff
@@ -746,6 +777,56 @@ function _buildDispositionSection(item) {
     return null;
 
   const d = item.disposition;
+  if (d.type === "traded" && Array.isArray(d.tradedForUuids) && d.tradedForUuids.length > 0) {
+    const section = _section("Trade");
+    const base = _buildDispositionSection({ ...item, disposition: { ...d, tradedForUuids: [] } });
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/viewModal.js]: **Stale — no longer applicable.** Path 1 (received-side acquired-via-trade view) and Path 2 (disposed-side trade summary) are now separate non-recursive code paths in `_buildDispositionSection`. The Path 1 early-return at the top of the function does not affect Path 2's rendering since they handle mutually exclusive item states (`tradedFromUuid` vs `tradedForUuids`).

File: js/viewModal.js [hunk 6]
```diff
@@ -737,6 +737,37 @@ function _mergeRetailHistoryEntries(itemRetailEntries, goldbackRetailEntries) {
...
+      });
+      line.appendChild(openBtn);
+    }
+    const unlinkBtn = _el("button", "view-footer-btn secondary trade-unlink-btn");
+    unlinkBtn.type = "button";
+    unlinkBtn.textContent = "Unlink from Trade";
+    unlinkBtn.addEventListener("click", () => {
+      if (source && typeof unlinkTradeItem === "function") unlinkTradeItem(source, item.uuid);
+      closeViewModal();
+    });
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/viewModal.js:856]: **Fixed in `97b98c59`** — added an `else` branch that clears `item.tradedFromUuid` directly (via `delete` + `saveInventory()`) when the source item is missing. `renderTable()` is also called to refresh the UI before `closeViewModal()`. All 17 disposition tests pass.

File: js/events.js [hunk 7]
```diff
@@ -4799,6 +4805,69 @@ if (removeItemDisposeBtn) {
...
+
+const tradeEls = () => ({
+  section: document.getElementById("tradeLinkSection"),
+  search: document.getElementById("tradeItemSearch"),
+  suggestions: document.getElementById("tradeItemSuggestions"),
+  linked: document.getElementById("tradeLinkedItems"),
+  summary: document.getElementById("tradeValueSummary"),
+});
+
+const getTradeDate = () => document.getElementById("dispositionDate")?.value || "";
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/events.js]: **Fixed in `97b98c59`** — `getTradeDate()` now falls back to today's local date (via `toLocaleDateString('en-CA')`) when `dispositionDate` is empty, so the trade value summary populates immediately when an item is selected. The dispose modal already auto-populates the date input (STRK-130), so this only matters in edge cases like programmatic test setup.

File: js/viewModal.js [hunk 8]
```diff
@@ -737,6 +737,37 @@ function _mergeRetailHistoryEntries(itemRetailEntries, goldbackRetailEntries) {
...
+    section.appendChild(line);
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/viewModal.js]: **Fixed in `97b98c59`** — same fix as the Gemini thread on this code. The unlink button now clears `item.tradedFromUuid` directly (and persists via `saveInventory()`) when the source item can't be found, so users can recover from orphaned back-references.

File: js/viewModal.js [hunk 9]
```diff
@@ -746,6 +777,56 @@ function _buildDispositionSection(item) {
...
+    summary.innerHTML = `<span>Current source value ${formatCurrency(sourceValue)}</span><strong class="trade-gain-loss">TRADE GAIN/LOSS ${(receivedTotal - sourceValue >= 0 ? "+" : "") + formatCurrency(receivedTotal - sourceValue)}</strong>`;
+    section.appendChild(summary);
+    const editBtn = _el("button", "view-footer-btn secondary");
+    editBtn.type = "button";
+    editBtn.textContent = "Edit Trade";
+    editBtn.addEventListener("click", () => {
+      closeViewModal();
+      const idx = inventory.findIndex((candidate) => candidate.uuid === item.uuid);
+      if (idx >= 0 && typeof openRemoveItemModal === "function") openRemoveItemModal(idx, true);
+    });
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/viewModal.js]: **Stale — no longer applicable.** The 'Edit Trade' inline editor in `_buildDispositionSection` pre-populates `editUuids = [...d.tradedForUuids]` from the existing disposition. Save preserves all values via `updateTradeLinks()`. The dispose modal is no longer involved in trade editing after the STRK-124 polish pass.

File: js/events.js [hunk 10]
```diff
@@ -4799,6 +4805,69 @@ if (removeItemDisposeBtn) {
...
+          </div>
+        </div>`;
+    })
+    .join("");
+  if (summary) summary.textContent = formatCurrency(total);
+  const dispositionAmount = document.getElementById("dispositionAmount");
+  if (dispositionAmount && total > 0 && !dispositionAmount.value) {
+    dispositionAmount.value = total.toFixed(2);
+  }
+};
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/events.js:4870]: **Stale — no longer applicable.** The per-item custom value input UI was removed during the STRK-124 polish pass. Current `renderPendingTradeLinks` uses chip pills without per-item value inputs. The single Amount field in the dispose modal is the user-editable trade value, and STRK-132 cost basis carryover uses the given-up item's melt at trade date.

File: js/events.js [hunk 11]
```diff
@@ -4799,6 +4805,69 @@ if (removeItemDisposeBtn) {
...
+      return `
+        <div class="trade-linked-item" data-uuid="${sanitizeHtml(uuid)}">
+          <div>
+            <strong>${sanitizeHtml(item.name || "Unnamed item")}</strong>
+            <div class="trade-item-meta">${sanitizeHtml(item.metal || "")} · Qty ${Number(item.qty) || 1}</div>
+          </div>
+          <div class="trade-linked-item__value">
+            <input type="number" class="trade-value-input" data-uuid="${sanitizeHtml(uuid)}" value="${meltValue ? meltValue.toFixed(2) : ""}" placeholder="—" />
+            <span class="trade-value-label">${value ? "spot value" : "custom"}</span>
+            <button type="button" class="btn secondary trade-linked-item__remove" data-remove-trade-uuid="${sanitizeHtml(uuid)}">×</button>
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/events.js]: **Stale — no longer applicable.** The `spot value/custom` label logic was removed with the per-item value input UI during STRK-124. There is no longer per-chip labeling — the trade Amount is a single field on the modal.

File: index.html [hunk 12]
```diff
@@ -8410,6 +8410,31 @@ <h2 id="removeItemModalTitle">Remove Item</h2>
...
+                <div id="tradeItemSuggestions" class="trade-item-suggestions" role="listbox"></div>
+              </div>
+              <div id="tradeLinkedItems" class="trade-linked-items" aria-live="polite"></div>
+              <button type="button" id="tradeAddNewItemBtn" class="btn secondary">
+                + Add new item to inventory
+              </button>
+              <div id="tradeEditControls" class="trade-edit-controls" style="display: none">
+                <button type="button" id="tradeSaveEditBtn" class="btn primary">Save trade</button>
+                <button type="button" id="tradeCancelEditBtn" class="btn secondary">Cancel</button>
+              </div>
```
[the remark this answers is not in the document]
Comment (lbruton) [reply index.html]: **Stale — no longer applicable.** Those IDs (`#tradeEditControls`, `tradeSaveEditBtn`, `tradeCancelEditBtn`) are not present in `index.html` or any JS file in the current codebase — verified via grep with zero matches. The 'Edit Trade' inline editor is DOM-constructed at runtime inside `viewModal.js._buildDispositionSection` Path 2, with handlers bound directly in JS rather than as static HTML.

File: css/styles.css [hunk 13]
```diff
@@ -969,6 +969,106 @@ section {
...
 }
+
+.trade-link-section,
+.trade-section {
+  display: grid;
+  gap: 0.75rem;
+  padding: 0.75rem;
+  border: 1px solid var(--border-color);
+  border-radius: 8px;
+  background: var(--bg-secondary);
```
[the remark this answers is not in the document]
Comment (lbruton) [reply css/styles.css]: **Fixed within the PR** — `.trade-section` was removed in commit `ba1d100f` (STRK-123: rework trade UI/UX to match approved mockup), which is already part of this branch. A current grep across all .html, .js, and .css files confirms zero references to `trade-section` in the codebase.

File: js/types.js [hunk 14]
```diff
@@ -39,8 +39,12 @@
  * @property {"auto"|"circle"|"rectangle"} [reverseImageFrame] - Optional reverse image frame override
  * @property {string|null} [obverseSharedImageId] - UUID of source item if obverse image was tagged from the shared library (null for original uploads)
  * @property {string|null} [reverseSharedImageId] - UUID of source item if reverse image was tagged from the shared library (null for original uploads)
+ * @property {string} [tradedFromUuid] - UUID of the disposed item that was traded to acquire this item
  * @property {boolean} [collectable] - Whether item is marked as collectable
  * @property {AttachmentEntry[]} [attachments] - Per-item file attachments (STRK-45)
+ * @property {Object} [disposition] - Disposition details for removed or disposed items
+ * @property {string[]} [disposition.tradedForUuids] - UUIDs of inventory items received in this trade
+ * @property {Object.<string, {meltValue: number, spotPrice: number, isCustom: boolean}>} [disposition.tradeValues] - Trade value snapshots keyed by received item UUID
```
[the remark this answers is not in the document]
Comment (lbruton) [reply js/types.js:47]: **Fixed in `97b98c59`** — widened `tradedFromUuid` JSDoc type from `{string}` to `{string|null}` to match the persistence-layer behavior (backup/restore may set to null when the source item is missing). Style follows the existing `{string|null}` pattern already used for the shared-image-id fields on the same typedef.
